### PR TITLE
Feature/structured metadata

### PR DESCRIPTION
**Pull request by marcpareoee**
Adds 

POST /project_structured_metadata

For creating one or more metadata items. Always completely replaces any existing data.

Also probably GET routes via DRF, though I didn't test em since we only use this API for loading data.

I fleshed out the models based on what was on the prod metadata tables. Any fields I should add/change?

### CODE REVIEW

**Comment by marcpareoee** [comment]
@philngo @daveyeager wanna give this a read?

File: datastore/models.py [hunk 1]
```diff
@@ -203,3 +203,36 @@ def get_traces(self):
...
+    project_structured_metadata = models.ForeignKey(ProjectStructuredMetadata, related_name='measures', on_delete=models.CASCADE)
+    cost = models.FloatField(null=True)
+    predicted_annual_savings = models.FloatField(null=True)
+    measure_type = models.CharField(max_length=256, null=True)
+
+    def __str__(self):
+        return ("project_id={}".format(self.project_structured_metadata.project.project_id))
+
+@python_2_unicode_compatible
+class Improvement(models.Model):
```
Comment (philngo) [inline datastore/models.py]: In my mind, "improvement" is another word for "measure". Unless there's a subtlety I'm missing, I think we could drop this model and just call everything a measure.

File: datastore/models.py [hunk 2]
```diff
@@ -203,3 +203,36 @@ def get_traces(self):
         )
         parsed_filter = parse_filter(self.filter_specification)
         return filtered_traces(parsed_filter)
+
+@python_2_unicode_compatible
+class ProjectStructuredMetadata(models.Model):
+    """Metadata with a schema for Projects."""
```
Comment (philngo) [inline datastore/models.py]: Other fields we could use right away:
- `cost` (total)
- `sqft`
- `climate_zone`
- `predicted_annual_baseline` (total)
- `predicted_annual_reporting` (total)
- `predicted_annual_savings` (total)

File: datastore/models.py [hunk 3]
```diff
@@ -203,3 +203,36 @@ def get_traces(self):
...
+
+    def __str__(self):
+        return ("project_id={}".format(self.project.project_id))
+
+@python_2_unicode_compatible
+class Measure(models.Model):
+    """Metadata with a schema for Projects."""
+    project_structured_metadata = models.ForeignKey(ProjectStructuredMetadata, related_name='measures', on_delete=models.CASCADE)
+    cost = models.FloatField(null=True)
+    predicted_annual_savings = models.FloatField(null=True)
```
Comment (philngo) [inline datastore/models.py]: I'd add `predicted_annual_baseline` and `predicted_annual_reporting`

**Comment by philngo** [comment]
Now that I think about it, those `predicted_annual_*` fields are still a bit ambiguous. Are they combined electricity/gas? If so, what units? If not, now do we represent multiple fuels? So breaking these down by fuel might be a good idea. I want to think about this a bit more before we merge this. The other fields are pretty straightforward.

**Comment by marcpareoee** [comment]
It's no big deal to revisit at a future date and add more/change the fields with migrations.

**Comment by philngo** [comment]
In that spirit, maybe the best thing to do is leave the prediction fields out for now (in both places) until we have a better idea of what we need.

**Comment by mcgeeyoung** [comment]
I'm going to be pushing more on this in the next couple of weeks. Predicted fields are going to be essential - that's how we get realization rate. But we should work on this from a more solid foundation than speculation.

**Comment by daveyeager** [comment]
Looks good to me - gives us good foundation to build on with future migrations.

**Pull request merged**